python/tests/plugins/test_gen_design_doc_impl.py

```diff
@@ -1,39 +1,50 @@
 import os
-from tempfile import TemporaryDirectory
 from unittest import TestCase
 
 from aac.plugins.gen_design_doc.gen_design_doc_impl import gen_design_doc
 
 from tests.helpers.assertion import assert_plugin_success
+from tests.helpers.io import temporary_test_file
 
 
 class TestGenerateDesignDocumentPlugin(TestCase):
     def test_can_generate_design_doc_with_models(self):
-        with TemporaryDirectory() as temp_dir:
-            test_model_file_name = f"{temp_dir}/test_model.yaml"
-            test_design_doc_file_name = None
-            with open(test_model_file_name, "w") as arch_file:
-                arch_file.write(TEST_MODEL)
-
-            result = gen_design_doc(test_model_file_name, temp_dir)
-            assert_plugin_success(result)
-
+        with temporary_test_file(TEST_MODEL) as test_model: 
+            temp_dir = os.path.dirname(test_model.name)
+            result = gen_design_doc(test_model.name, temp_dir)
+            assert_plugin_success(result) 
             files = os.listdir(temp_dir)
             self.assertEqual(len(files), 2)
-
             test_design_doc_file_name, *_ = [
-                f for f in files if f != os.path.basename(test_model_file_name)
+                f for f in files if f != os.path.basename(test_model.name)
             ]
-            with open(f"{temp_dir}/{test_design_doc_file_name}", "r") as design_doc:
-                markdown = design_doc.read()
-                self.assert_headings(markdown)
+            with open(os.path.join(temp_dir, test_design_doc_file_name)) as markdown_file:
+                document_title = os.path.basename(test_model.name)
+                markdown = markdown_file.read()
+                self.assert_headings(markdown, document_title)
                 self.assert_schema(markdown)
                 self.assert_model(markdown)
                 self.assert_use_case(markdown)
+   
+    def test_can_handle_names_with_dots(self):
+        with temporary_test_file(TEST_MODEL_2) as test_yaml:
+            result = gen_design_doc(test_yaml.name, os.path.dirname(test_yaml.name))
+            assert_plugin_success(result)
+            
+            temp_dir = os.path.dirname(test_yaml.name)
+            files = os.listdir(temp_dir)
+            self.assertEqual(len(files), 2)
+    
+            test_design_doc_file_name, *_ = [
+                f for f in files if f != os.path.basename(test_yaml.name)
+            ]
```

Comment:
Please also update this on lines 18-20 as I can't leave it as a suggestion.

```suggestion
            test_design_doc_file_name, *_ = [f for f in files if f != os.path.basename(test_yaml.name)]
```

Reply:
Please check the latest commit. I made the above change to the first and second test